apps/dev/src/commands/feeds/read/adfs.ts

```diff
@@ -9,9 +9,43 @@ import {
 import { renderTui, drawTable } from '@blocksense/base-utils/tty';
 import { listEvmNetworks, readEvmDeployment } from '@blocksense/config-types';
 import { AggregatedDataFeedStoreConsumer } from '@blocksense/contracts/viem';
+import { formatTimestamp } from '../../utils';
 
 const availableNetworks = await listEvmNetworks();
 
+export const parseADFSData = (hexData: string) => {
+  const cleanHex = hexData.startsWith('0x') ? hexData.slice(2) : hexData;
```

Comment:
We can add a check if the passed data is of the correct length and can be converted to a numeric value and a timestamp.
```suggestion
  const cleanHex = hexData.startsWith('0x') ? hexData.slice(2) : hexData;
  if (cleanHex.length !== 64) {
    throw new Error(
      'Unexpected ADFS data length, expected 32 bytes (64 hex characters), got ' +
        cleanHex.length,
    );
  }
```